src/sqlparser/src/ast/mod.rs

```diff
@@ -2953,6 +2954,48 @@ impl fmt::Display for SqlOption {
     }
 }
 
+impl TryFrom<(&String, &String)> for SqlOption {
+    type Error = ParserError;
+
+    fn try_from((name, value): (&String, &String)) -> Result<Self, Self::Error> {
+        let mut tokenizer = Tokenizer::new(value);
+        let tokens = tokenizer.tokenize_with_location()?;
+        let mut parser = Parser(&tokens);
+
+        let value = {
+            const CONNECTION_REF_KEY: &str = "connection";
+            if name.eq_ignore_ascii_case(CONNECTION_REF_KEY) {
+                let connection_name = parser.parse_object_name().map_err(|e| {
+                    ParserError::ParserError(format!(
+                        "Failed to parse connection name in connection option: {}",
+                        e
+                    ))
+                })?;
+                let connection_name = match connection_name.0.as_slice() {
+                    [ident] if ident.real_value() == CONNECTION_REF_KEY => {
+                        parser.parse_object_name().map_err(|e| {
+                            ParserError::ParserError(format!(
+                                "Failed to parse connection name in connection option: {}",
+                                e
+                            ))
+                        })?
+                    }
+                    _ => connection_name,
+                };
+                SqlOptionValue::ConnectionRef(ConnectionRefValue { connection_name })
+            } else {
+                parser.parse_value_and_obj_ref::<false>().map_err(|e| {
+                    ParserError::ParserError(format!("Failed to parse value in option {}", e))
+                })?
+            }
+        };
```

Comment:
maybe a extract it into a individual func and share usage with `parse_sql_option`  because we do some spec logic for connection.